PA193_mnemonic_Slytherin/mnemonic.py

```diff
@@ -24,7 +28,14 @@ def __generate_seed(mnemonic: str, seed_password: str = '') -> bytes:
     :rtype: bytes
     :return: Seed
     """
-    pass
+    #who must check the encoding of both inputs? should be UTF-8 NFKD
+    #where should I catch UnicodeError?
+    mnemonic = mnemonic.encode()  #encoding string into bytes, UTF-8 by default
```

Comment:
> The wordlist can contain native characters, but they must be encoded in UTF-8 using Normalization Form Compatibility Decomposition (NFKD). 
> https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki#wordlist

I'm not sure if I understand now, when do you expect UnicodeError to happen?

Reply:
If the correctness of the input (especially, the password) is guaranteed, then no problem.

Reply:
When we are encoding UTF-8 as bytes, are there posibilities of UnicodeError? We are not trying to get ASCII characters, just values of those bytes.

Reply:
Do we need to be careful about UTF-8 NFKD? Could we have mnemonic in text file which would crash it?

Reply:
Actually, I wasn't able to crash the script, maybe with str to bytes it's nearly impossible to do. But we still have problems if our wordlist file isn't UTF-8 encoded (the same passwords/mnemonics with different encoding produce different results, displaying problem, etc.). That's relevant for non-Latin dictionaries, though.

Reply:
Both mnemonic and dictionary must be UTF-8 according to BIP39. If a dicionary isn't in valid UTF-8, then it's invalid dictionary. Same for mnemonic, if it isn't UTF-8, then we should reject it.